src/view/DashboardView.java

```diff
@@ -0,0 +1,202 @@
+package view;
+
+import interface_adapter.dashboard.DashboardState;
+import interface_adapter.dashboard.DashboardViewModel;
+import interface_adapter.dashboard.ProjectData;
+import interface_adapter.add_project.AddProjectController;
+
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.text.ParseException;
+import java.util.ArrayList;
+import javax.swing.text.MaskFormatter;
+
+public class DashboardView extends JFrame implements PropertyChangeListener {
+    private final DashboardViewModel dashboardViewModel;
+    private JPanel dashboardPanel;
+    private ArrayList<ProjectData> projectsList;
+
+    public DashboardView(DashboardViewModel dashboardViewModel, AddProjectController addProjectController) {
+        this.dashboardViewModel = dashboardViewModel;
+
```

Comment:
I think viewName attribute would be nice, like public final String viewName = "Dashboard View" for case when we need to switch between views using JPanel in Main.

Reply:
Great catch Daniel! I have implemented this and will be pushed.